cve_bin_tool/helper_script.py

```diff
@@ -0,0 +1,185 @@
+        self.logger = LOGGER.getChild(self.__class__.__name__)
+        self.product = product_name or self.parse_filename(filename)
+        self.vendor = vendor_name
+
+        # for setting the database
+        self.connection = None
+        self.dbpath = os.path.join(DISK_LOCATION_DEFAULT, DBNAME)
+
+        # for extraction
+        self.walker = DirWalk().walk
+        self.file_stack = []  # stores extracted filepaths
+
+        # for output (would use in future)
+        self.contain_patterns = []
+        self.filename_pattern = []
+        self.version_pattern = []
+        self.vendor_product = []
+
+    def extract_and_parse_file(self):
+        """extracts and parses the file for common patterns, version strings and common filename patterns"""
+        with self.extractor as ectx:
+            if ectx.can_extract(self.filename):
+                binary_string_list = []
+                for filepath in self.walker([ectx.extract(self.filename)]):
+                    clean_path = VersionScanner.clean_file_path(filepath)
+                    self.file_stack.append(f"{clean_path}")
+                    # self.logger.debug(f"scanning binaries from {clean_path}")
+
+                    # see if the file is ELF binary file and parse for strings
+                    if self.is_executable(filepath):
+                        matches = self.search_pattern(self.parse_strings(filepath))
+                        binary_string_list.append(matches)
+                        self.filename_pattern.append(filepath.split("/")[-1])
+                        # self.logger.debug(f"{self.filename_pattern}")
+
+                return binary_string_list
+
+    def is_executable(self, filename):
+        """checks if given file is executable/ELF binary file"""
+
+        if inpath("file"):
+            # using system file if available (for performance reasons)
+            o = subprocess.check_output(["file", filename])
+            o = o.decode(sys.stdout.encoding)
+
+            if (
+                ("LSB " not in o)
+                and ("LSB shared" not in o)
+                and ("LSB executable" not in o)
+                and ("PE32 executable" not in o)
+                and ("PE32+ executable" not in o)
+                and ("Mach-O" not in o)
+            ):
+                return False
+        # else using python implementation of file
+        elif not is_binary(filename):
+            return False
+
+        return True
+
+    def parse_strings(self, filename):
+        """parse binary file's strings"""
+
+        # self.logger.debug(f"running for strings over {filepath}")
+        if inpath("strings"):
+            # use "strings" on system if available (for performance)
+            o = subprocess.check_output(["strings", filename])
+            lines = o.decode("utf-8").splitlines()
+        else:
+            # Otherwise, use python implementation
+            s = Strings(filename)
+            lines = s.parse()
+        return lines
```

Comment:
I'm reusing these from `scan_file()` in `VersionScanner()`. I've sent a pr, breaking down `scan_file()` accordingly (I would be updating this once #1185 gets merged or it's decided what should be done about this...).

Reply:
Awesome.  let's merge #1185 first and then refactor this.